### PR TITLE
Add R bin path to PATH environment variable.

### PR DESCRIPTION
**Pull request by karthiknadig**
https://github.com/Microsoft/RTVS/issues/3592

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -183,6 +183,35 @@ namespace rhost {
...
     }
 
+    void add_dir_to_loader_path(fs::path &r_dir) {
+#ifdef _WIN32
+        wchar_t buff[32767] = {}; // max buffer size for environment variable values on windows.
+        int nchars = GetEnvironmentVariable(L"PATH", buff, ARRAYSIZE(buff));
+        HRESULT err = S_OK;
+        if (nchars == 0) {
+            err = HRESULT_FROM_WIN32(GetLastError());
+            logf(log_verbosity::minimal, "Failed to get PATH environment variable [GetEnvironmentVariable]: %x\n", err);
```
Comment (MikhailArkhipov) [inline src/main.cpp]: It is, of course, unusual, but technically PATH variable doesn't have to exist...

File: src/main.cpp [hunk 2]
```diff
@@ -183,6 +183,35 @@ namespace rhost {
...
+
+        std::wstring old_path_env(buff);
+        std::wstring new_path_env = r_dir.make_preferred().wstring() + L";" + old_path_env;
+        if (SetEnvironmentVariableW(L"PATH", new_path_env.c_str()) == 0) {
+            err = HRESULT_FROM_WIN32(GetLastError());
+            logf(log_verbosity::minimal, "Failed to set PATH environment variable [SetEnvironmentVariable]: %x\n", err);
+            std::terminate();
+        }
+#else
+        std::string old_path_env(std::getenv("LD_LIBRARY_PATH"));
```
Comment (MikhailArkhipov) [inline src/main.cpp]: If it doesn't exist, does it yield null or empty string?

File: src/main.cpp [hunk 3]
```diff
@@ -183,6 +183,35 @@ namespace rhost {
...
+        std::string new_path_env = r_dir.make_preferred().string() + ":" + old_path_env;
+        if (setenv("LD_LIBRARY_PATH", new_path_env.c_str(), 1) == -1) {
```
Comment (MikhailArkhipov) [inline src/main.cpp]: Out of curiosity - `setenv` doesn't work in Win32?

Comment (karthiknadig) [inline src/main.cpp]: `setenv` is not supported in Win32.

File: src/main.cpp (hunk 1 above)
Comment (karthiknadig) [inline src/main.cpp]: You are correct. I made an assumption that "PATH" is always available.

File: src/main.cpp (hunk 2 above)
Comment (karthiknadig) [inline src/main.cpp]: I made an assumption that "PATH" is always available.

File: src/main.cpp [hunk 4]
```diff
@@ -183,6 +183,40 @@ namespace rhost {
...
+#ifdef _WIN32
+        wchar_t buff[32767] = {}; // max buffer size for environment variable values on windows.
+        int nchars = GetEnvironmentVariable(L"PATH", buff, ARRAYSIZE(buff));
+        HRESULT err = S_OK;
+        if (nchars == 0) {
+            err = HRESULT_FROM_WIN32(GetLastError());
+            if (err == HRESULT_FROM_WIN32(ERROR_ENVVAR_NOT_FOUND)) {
+                buff[0] = '\0';
+            } else {
+                logf(log_verbosity::minimal, "Failed to get PATH environment variable [GetEnvironmentVariable]: %x\n", err);
```
Comment (int19h) [inline src/main.cpp]: `fatal_error` will log & terminate.

**Pull request merged**